contentcuration/contentcuration/utils/publish.py

```diff
@@ -945,6 +967,72 @@ def fill_published_fields(channel, version_notes):
             }
         }
     )
+
+    # Calculate non-distributable licenses (All Rights Reserved)
+    all_rights_reserved_id = (
+        ccmodels.License.objects.filter(license_name=licenses.ALL_RIGHTS_RESERVED)
+        .values_list("id", flat=True)
+        .first()
+    )
+
+    non_distributable_licenses = (
+        [all_rights_reserved_id]
+        if all_rights_reserved_id and all_rights_reserved_id in license_list
+        else []
+    )
+
+    # records for each unique description so reviewers can approve/reject them individually.
+    # This allows centralized tracking of custom licenses across all channels.
+    special_permissions_id = (
+        ccmodels.License.objects.filter(license_name=licenses.SPECIAL_PERMISSIONS)
+        .values_list("id", flat=True)
+        .first()
+    )
+
+    if special_permissions_id and special_permissions_id in license_list:
+        special_perms_descriptions = list(
+            published_nodes.filter(license_id=special_permissions_id)
+            .exclude(license_description__isnull=True)
+            .exclude(license_description="")
+            .values_list("license_description", flat=True)
+            .distinct()
+        )
+
+        if special_perms_descriptions:
+            new_licenses = [
+                ccmodels.AuditedSpecialPermissionsLicense(
+                    description=description, distributable=False
+                )
+                for description in special_perms_descriptions
+            ]
+
+            ccmodels.AuditedSpecialPermissionsLicense.objects.bulk_create(
+                new_licenses, ignore_conflicts=True
+            )
+
+    if channel.version_info:
+        channel.version_info.resource_count = channel.total_resource_count
+        channel.version_info.kind_count = kind_counts
+        channel.version_info.size = int(channel.published_size)
+        channel.version_info.date_published = channel.last_published
+        channel.version_info.version_notes = version_notes
+        channel.version_info.included_languages = language_list
```

Comment:
While checking how the publish workflow works with the `full_clean` call, I noticed that the `language_list` included some "null" values as items. And this is because in [this](https://github.com/AlexVelezLl/studio/blob/33e039183aa4c8b06af3cacb2ed2f2860033138c/contentcuration/contentcuration/utils/publish.py#L927) line we are missing the `.exclude(files__language=None)`, it is clearly not a regression of this PR, but could you fix it, please? Thanks!